### PR TITLE
Fix/mobl 602 optional sdk variables

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
Set SDK variables to nullable/optional to avoid SDK crashes if SDK in not initalised
Added api key checks for custom event, tracking events and live content
Removed live content duplicate implementation
Removed unused variable deviceToken from Blueshift.h
Removed registerForNotification call from the validateChangeInUNAuthorizationStatus as its not required
Improved error logging

### CODE REVIEW

File: BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m [hunk 1]
```diff
@@ -15,53 +15,56 @@
...
+            [pushTrackParametersMutableDictionary setObject:urlElement forKey: kNotificationURLElementKey];
+        }
+        if([[pushDetailsDictionary objectForKey: kNotificationTypeIdentifierKey] isEqualToString:kNotificationKey] && [self isNotNilAndNotEmpty:pushDeepLinkURL]) {
+            NSString *encodedUrl = [BlueShiftInAppNotificationHelper getEncodedURLString:pushDeepLinkURL];
+            if (encodedUrl) {
+                [pushTrackParametersMutableDictionary setObject:encodedUrl forKey: kNotificationURLElementKey];
+            }
+        }
+        if (deviceId) {
+            [pushTrackParametersMutableDictionary setObject:deviceId forKey: @"device_id"];
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m:59]: I saw you adding constants for these two strings in a previous PR. can we use them here?

Comment (rahulrvp) [reply BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m:59]: for device_id and app_name

File: BlueShift-iOS-SDK/BlueShift.m [hunk 2]
```diff
@@ -191,8 +191,8 @@ - (void)setDeviceToken {
 
 - (NSString *) getDeviceToken {
     NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
-    _deviceToken = (NSString *)[defaults objectForKey:kBlueshiftDeviceToken];
-    return _deviceToken;
+    NSString* deviceToken = (NSString *)[defaults objectForKey:kBlueshiftDeviceToken];
+    return deviceToken;
```
[the remark this answers is not in the document]
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShift.m:195]: I had added like this to make the debugging easy. With just return statement it's hard to get the value from it.

File: BlueShift-iOS-Extension-SDK/BlueshiftExtensionAnalyticsHelper.m [hunk 3]
```diff
@@ -10,36 +10,37 @@
...
+        }
+        if (sdkVersion) {
+            [pushTrackParametersMutableDictionary setObject:sdkVersion forKey: kInAppNotificationModalSDKVersionKey];
+        }
+        if (element) {
+            [pushTrackParametersMutableDictionary setObject:element forKey: kInAppNotificationModalElementKey];
+        }
+        if (timestamp) {
+            [pushTrackParametersMutableDictionary setObject:timestamp forKey: kInAppNotificationModalTimestampKey];
+        }
```
[the remark this answers is not in the document]
Comment (ketanshikharebsft) [reply BlueShift-iOS-Extension-SDK/BlueshiftExtensionAnalyticsHelper.m:42]: Thanks for the suggestion, I can definitely take this up in the next release as this is going to be a big change.

File: BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m (hunk 1 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m]: fixed.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**